src/sql/workbench/contrib/editData/browser/editDataEditor.ts

```diff
@@ -748,7 +748,7 @@ export class EditDataEditor extends EditorPane {
 				editDataInput.savedViewState = this._sqlEditor.getControl().saveViewState();
 			}
 			if (editDataInput.results) {
-				editDataInput.results.onSaveViewStateEmitter.fire();
+				editDataInput.results.onSaveViewStateEmitter.fire(editDataInput.results.isDisposed());
```

Comment:
correct way to use event is to unsubscribe instead of passing in isDisposed value and use it as a condition check.